ops_boba/cloudformation/11-infrastructure-transaction-service-gnosis-mainnet.yaml

```diff
@@ -0,0 +1,181 @@
+                  - "elasticloadbalancing:DescribeTargetGroups"
+                  - "elasticloadbalancing:DescribeTargetHealth"
+                  - "elasticloadbalancing:RegisterTargets"
+                  - "elasticfilesystem:ClientMount"
+                  - "elasticfilesystem:ClientRootAccess"
+                  - "elasticfilesystem:ClientWrite"
+                  - "elasticfilesystem:DescribeMountTargets"
+                Resource: "*"
+
+  EcsInstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Roles: [!Ref EcsInstanceRole]
+
+  EcsSecurityGroupGnosis:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: !Sub "SG for ${AWS::StackName} Landscape ECS Cluster Gnosis"
+      VpcId:
+        Fn::ImportValue: !Sub ${InfrastructureStackName}:VpcId
+      SecurityGroupIngress:
+      - Description: Allow all traffic
+        IpProtocol: "-1"
+        CidrIp: 0.0.0.0/0
+      Tags:
+        - Key: Name
+          Value: !Sub "${AWS::StackName}-Gnosis-ECS-Cluster"
+
+  EcsLaunchConfigurationGnosis:
+    Type: AWS::AutoScaling::LaunchConfiguration
+    Metadata:
+      AWS::CloudFormation::Init:
+        config:
+          files:
+            '/etc/ecs/ecs.config':
+              content: !Sub |
+                ECS_CLUSTER=${EcsCluster}
+                ECS_ENABLE_TASK_IAM_ROLE=true
+                ECS_CONTAINER_STOP_TIMEOUT=5s
+              mode: "000644"
+              owner: root
+              group: root
+    Properties:
+      InstanceType: !Ref EcsInstanceType
+      SecurityGroups:
+      - !Ref EcsSecurityGroupGnosis
+      IamInstanceProfile: !Ref EcsInstanceProfile
+      ImageId: !Ref EcsImageId
+      UserData:
+        Fn::Base64: !Sub |
+            #!/bin/bash -ex
+            yum -y install aws-cfn-bootstrap amazon-ssm-agent amazon-efs-utils
+            systemctl enable --now amazon-ssm-agent
+            echo "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDbEgniumTWAjZjw7vgcMCrHALO7KcjXHo3I09ulp7ufd1jpk/0PCAPwRqURLZWujjkB/OGto8W/YqCF7F5s3VJvC5lq0Sbbmagyz7obTYRVBNA656wdw2T6udHoci26kQEJoEZCHjlnCnPnNC1IkiJMwjbXivcVdUeCuFaUREWqOB5qPDEoG3xfKqHF1qlVhuuI/Q14AsgUGtMNKQyuR97k/pL2KncTTIg1Ev2pMm3/4sND5TFtX907LNEz1fu1JQqpM/fXUM/a+rlREaBj+2+nWC37+TFjd3WWvohdMEljRGH8JnTcYU5hvfdZV62pLgNjz0LPAn+tLugwacbGy6J chenboyuan@Macbook.hsd1.va.comcast.net" > /home/ec2-user/.ssh/authorized_keys
+            mkdir -p /mnt/efs /mnt/efs/db
+            /opt/aws/bin/cfn-init -v --stack ${AWS::StackName} --resource EcsLaunchConfigurationGnosis --region ${AWS::Region}
+            /opt/aws/bin/cfn-signal -e $? --stack ${AWS::StackName} --resource AutoScalingGroupGnosis --region ${AWS::Region} || true
+  AutoScalingGroupGnosis:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    Properties:
+      LaunchConfigurationName: !Ref EcsLaunchConfigurationGnosis
+      MetricsCollection:
+      - Granularity: 1Minute
+        Metrics:
+        - GroupTotalInstances
+      MinSize: 1
+      MaxSize: 1
+      DesiredCapacity: 1
```

Comment:
Since this is mainnet, is there a need to consider autoscaling? Or is the application very stateful?

Reply:
Very stateful!